main/src/ca/uwaterloo/flix/language/ast/KindedAst.scala

```diff
@@ -317,7 +318,7 @@ object KindedAst {
 
   case class RelationalChoiceRule(pat: List[KindedAst.RelationalChoicePattern], exp: KindedAst.Expression)
 
-  case class RestrictableChoiceRule(pat: KindedAst.RestrictableChoicePattern, exp: KindedAst.Expression)
+  case class RestrictableChoiceRule(pat: KindedAst.RestrictableChoicePattern, sym: Option[Symbol.RestrictableCaseSym], exp: KindedAst.Expression)
```

Comment:
What's this symbol? is it for `Of` or?

Reply:
Yeah. We also talked about two case classes. The idea is just that this explicit tag is more straightforward once we do more intelligent tag detection (like supporting case bodies such as `let x = 42; Some(42)`)

Reply:
Is it worth more intelligent tag detection, if we have `Of`?

Reply:
Even if we don't have intelligent detection down the line, this style is just easier to manage since we don't have match on the structure of the expression.